Assets/MRTK/SDK/Experimental/Drawstring/Scripts/ElasticSystem.cs

```diff
@@ -0,0 +1,196 @@
+    {
+        /// <value>
+        /// Represents the lower bound of the extent,
+        /// specified as the norm of the n-dimensional extent
+        /// </value>
+        [SerializeField]
+        public float minStretch;
+
+        /// <value>
+        /// Represents the upper bound of the extent,
+        /// specified as the norm of the n-dimensional extent
+        /// </value>
+        [SerializeField]
+        public float maxStretch;
+
+        /// <value>
+        /// Whether the system, when approaching the upper bound,
+        /// will treat the upper bound as a snap point.
+        /// </value>
+        [SerializeField]
+        public bool snapToMax;
+
+        /// <value>
+        /// Points inside the extent to which the system will snap.
+        /// </value>
+        [SerializeField]
+        public T[] snapPoints;
+    }
+
+    /// <summary>
+    /// Properties of the damped harmonic oscillator differential system.
+    /// </summary>
+    [Serializable]
+    public struct ElasticProperties
+    {
+        /// <value>
+        /// Mass of the simulated oscillator element
+        /// </value>
+        [SerializeField]
+        public float mass;
+        /// <value>
+        /// Hand spring constant
+        /// </value>
+        [SerializeField]
+        public float hand_k;
+        /// <value>
+        /// End cap spring constant
+        /// </value>
+        [SerializeField]
+        public float end_k;
+        /// <value>
+        /// Snap point spring constant
+        /// </value>
+        [SerializeField]
+        public float snap_k;
+        /// <value>
+        /// Extent at which snap points begin forcing the spring.
+        /// </value>
+        [SerializeField]
+        public float snap_radius;
+        /// <value>
+        /// Drag/damper factor, proportional to velocity.
+        /// </value>
+        [SerializeField]
+        public float drag;
+    }
+
+    /// <summary>
+    /// Represents a damped harmonic oscillator over an
+    /// N-dimensional vector space, specified by generic type T.
+    /// 
+    /// This extensibility allows not just for 1, 2, and 3-D springs, but
+    /// allows for 4-dimensional quaternion springs.
+    /// </summary>
+    internal abstract class ElasticSystem<T>
+    {
+        protected ElasticExtentProperties<T> extentInfo;
+        protected ElasticProperties elasticProperties;
+
+        protected T currentValue;
+        protected T currentVelocity;
+
+        public ElasticSystem(T initialValue, T initialVelocity,
+                                ElasticExtentProperties<T> extentInfo, ElasticProperties elasticProperties)
+        {
+            this.extentInfo = extentInfo;
+            this.elasticProperties = elasticProperties;
+            currentValue = initialValue;
+            currentVelocity = initialVelocity;
+        }
+
+        /// <summary>
+        /// Update the internal state of the damped harmonic oscillator, given the forcing/desired value.
```

Comment:
This comment seems to be add odds with the fact that this function returns a value - if this purely were around updating internal state, I would assume this would be a void type.

If instead this is supposed to be about returning the new state, then we should update the language to specify that.

Reply:
Good point, fixed.